doc/pveclibmaindox.h

```diff
@@ -1104,59 +1215,1050 @@ s0 = vec_addeq (&c0, a0, b0, c1);
+* so libtool would allow target specific CFLAGS. So we can eliminate
+* libpvecPWR9.la, libpvecPWR8.la, and libpvecPWR7.la from
+* lib_LTLIBRARIES.
+*
+* Continuing the theme of separating the static archive elements from
+* DSO elements we rename libpveccommon.la to libpvecstatic.la.
+* We can add the common (none target specific) source files and CFLAGS
+* to <I>libpvecstatic_la</I>.
+* \code
+libpvecstatic_la_SOURCES = tipowof10.c decpowof2.c
+
+libpvecstatic_la_CFLAGS = $(AM_CPPFLAGS) $(PVECLIB_DEFAULT_CFLAGS) $(AM_CFLAGS)
+* \endcode
+* We still need to add the target specific objects generated by the
+* rules above to the libpvecstatic.a archive.
+* \code
+# libpvecstatic_la already includes tipowof10.c decpowof2.c.
+# Now add the name qualified -mcpu= target runtimes.
+libpvecstatic_la_LIBADD = vec_staticrt_PWR9.lo
+libpvecstatic_la_LIBADD += vec_staticrt_PWR8.lo
+libpvecstatic_la_LIBADD += vec_staticrt_PWR7.lo
+* \endcode
+* \note the libpvecstatic archive will contain 2 or 3 implementations
+* of each target specific function
+* (i.e. the function vec_mul128x128() will have implementations
+* vec_mul128x128_PWR7() and vec_mul128x128_PWR8(),
+* vec_mul128x128_PWR9()).
+* This OK because because the target suffix insures the name is unique
+* within the archive.
+* When an application calls function with the appropriate target suffix
+* (using the __VEC_PWR_IMP() wrapper macro) and links to libpvecstatic,
+* the linker will extract only the matching implementations and include
+* them in the static program image.
+*
+*
+* \subsection main_libary_issues_0_0_2 Building dynamic runtime libraries
+*
+* Building objects for dynamic runtime libraries is a bit more
+* complicated than building static archives. For one dynamic libraries
+* requires position independent code (<B>PIC</B>) while static code
+* does not. Second we want to leverage the Dynamic Linker/Loader's
+* GNU Indirect Function (See: <a href="https://gcc.gnu.org/onlinedocs/">
+* What is an indirect function (IFUNC)?</a>) binding mechanism.
+*
+* PIC functions may require a more complicated call linkage or
+* function prologue. This usually require the -fpic compiler option.
+* This is the case for the OpenPOWER ELF V2 ABI. Any PIC function must
+* assume that the caller may be from an different execution unit
+* (library or main executable).
+* So the called function needs to establish the Table of Contents
+* (<B>TOC</B>) base address for itself. This is the case if
+* the called function needs to reference static or const storage
+* variables or calls to functions in other dynamic libraries.
+* So it is normal to compile library runtime codes separately for
+* static archives and DSOs.
+*
+* \note The details of how the <B>TOC</B> is established differs
+* between the ELF V1 ABI (Big Endian POWER) and the
+* ELF V2 ABI (Little Endian POWER). This should not be
+* an issue if compile options (-fpic) are used correctly.
```

Comment:
How important is it to discuss the TOC in this file?

Reply:
Well the TOC switch handling IS WHY PIC is more complicated then static.

Let me think about ...